### PR TITLE
Adds render cross reference in links

### LINKED ISSUE #2: How to map a render service
**Problem:**

The render extension provides a mechanism to use a render service and it defines a key and a tile.
The web-map-links extensions provides links to e.g. xyz services.

How can these two extensions join efforts to map a render definition to a "render" service?

**Suggested solution**

Extend the link with a `key=value` that maps the `xyz` link to a render element:

```json
        {
          "rel": "xyz",
          "href": "https://view.iride-datalake.terradue.com/collections/sentinel-s2-l2a-cogs/items/S2B_MSIL2A_20231107T102139_N0509_R065_T32TNR_20231107T132337/tiles/WebMercatorQuad/{z}/{x}/{y}@1x?&assets=red&assets=green&assets=blue&color_formula=Gamma%20RGB%201.5%20Saturation%201%20Sigmoidal%20RGB%2010%200.3&rescale=1,6000&rescale=1,6000&rescale=1,6000",
          "type": "image/png",
          "title": "True colour composite visualized through a XYZ",
          "render": "rgb"
        },
```

where `rgb` in `"render": "rgb"` is a `render` definition

### PR DESCRIPTION
**Pull request by emmanuelmathot**
This PR Adds render cross reference in links and fix #2

### CODE REVIEW

File: json-schema/schema.json [hunk 1]
```diff
@@ -34,6 +34,37 @@
               }
             }
           }
+        },
+        {
+          "if": {
```
Comment (abarciauskas-bgse) [inline json-schema/schema.json:39]: Is this schema saying if stac extensions includes both renders and web-map-links, a render sub-property is required? Why do you think it should be required? Seems like an optional property.

Comment (abarciauskas-bgse) [reply json-schema/schema.json:39]: Also, I may be mis-reading the schema but does the then statement mean that there must be a link object in the links list that has "rel": "render" - in which case does the example above needs updating to include a link with "rel": "render"?

**Comment by abarciauskas-bgse** [comment]
@emmanuelmathot this makes sense to me but I wonder if this spec should be in the web-map-links extension rather than here, since the change is in the links object, not the renders object.

**Comment by emmanuelmathot** [comment]
> @emmanuelmathot this makes sense to me but I wonder if this spec should be in the web-map-links extension rather than here, since the change is in the links object, not the renders object.

The change specifies an extension to link but specific to render so this should be set here. If the logic discussed here: https://github.com/radiantearth/stac-spec/issues/1284 similar to this proposition changes, we will adapt to the logic but for the moment, we will add the possibility to link a render in a link.

**Pull request merged**